sdk/python/kfp/components/types/custom_artifact_types_test.py

```diff
@@ -0,0 +1,595 @@
+from kfp.components.types import type_annotations
+from kfp.components.types.artifact_types import Artifact
+from kfp.components.types.artifact_types import Dataset
+from kfp.components.types.type_annotations import Input
+from kfp.components.types.type_annotations import InputPath
+from kfp.components.types.type_annotations import Output
+from kfp.components.types.type_annotations import OutputPath
+import typing_extensions
+import yaml
+
+Alias = Artifact
+artifact_types_alias = artifact_types
+
+
+class TestFuncToRootAnnotationSymbol(unittest.TestCase):
+
+    def test_no_annotations(self):
+
+        def func(a, b):
+            pass
+
+        actual = custom_artifact_types.func_to_root_annotation_symbols_for_artifacts(
+            func)
+        expected = {}
+        self.assertEqual(actual, expected)
+
+    def test_no_return(self):
+
+        def func(a: int, b: Input[Artifact]):
+            pass
+
+        actual = custom_artifact_types.func_to_root_annotation_symbols_for_artifacts(
+            func)
+        expected = {'b': 'Artifact'}
+        self.assertEqual(actual, expected)
+
+    def test_with_return(self):
+
+        def func(a: int, b: Input[Artifact]) -> int:
+            return 1
+
+        actual = custom_artifact_types.func_to_root_annotation_symbols_for_artifacts(
+            func)
+        expected = {'b': 'Artifact'}
+        self.assertEqual(actual, expected)
+
+    def test_multiline(self):
+
+        def func(
+            a: int,
+            b: Input[Artifact],
+        ) -> int:
+            return 1
+
+        actual = custom_artifact_types.func_to_root_annotation_symbols_for_artifacts(
+            func)
+        expected = {'b': 'Artifact'}
+        self.assertEqual(actual, expected)
+
+    def test_alias(self):
+
+        def func(a: int, b: Input[Alias]):
+            pass
+
+        actual = custom_artifact_types.func_to_root_annotation_symbols_for_artifacts(
+            func)
+        expected = {'b': 'Alias'}
+        self.assertEqual(actual, expected)
+
+    def test_long_form_annotation(self):
+
+        def func(a: int, b: Input[artifact_types.Artifact]):
+            pass
+
+        actual = custom_artifact_types.func_to_root_annotation_symbols_for_artifacts(
+            func)
+        expected = {'b': 'artifact_types'}
+        self.assertEqual(actual, expected)
+
+    def test_named_tuple(self):
+
+        def func(
+            a: int,
+            b: Input[Artifact],
+        ) -> typing.NamedTuple('MyNamedTuple', [('a', int), (
+                'b', Artifact), ('c', artifact_types.Artifact)]):
+            InnerNamedTuple = typing.NamedTuple(
+                'MyNamedTuple', [('a', int), ('b', Artifact),
+                                 ('c', artifact_types.Artifact)])
+            return InnerNamedTuple(a=a, b=b, c=b)  # type: ignore
+
+        actual = custom_artifact_types.func_to_root_annotation_symbols_for_artifacts(
+            func)
+        expected = {'b': 'Artifact'}
```

Comment:
how should I interpret this result? Is it from the input or output? Should named tuple or single return be "processed" by `func_to_root_annotation_symbols_for_artifacts` at all?

Reply:
Will clarify this test and add handling for single and named tuple return annotations.